keps/sig-apps/20190226-maxunavailable-for-statefulsets.md

```diff
@@ -48,46 +48,47 @@ The purpose of this enhancement is to implement maxUnavailable for StatefulSet d
 will proceed `maxUnavailable` number of pods at a time. Note, that maxUnavailable does not affect podManagementPolicy which
-is only applicable during scaling.
+is only applicable during StatefulSet creation, deletion and scaling.
 
 
 ## Motivation
 
 Consider the following scenarios:-
 
 1. My containers publish metrics to a time series system. If I am using a Deployment, each rolling update creates a new pod name and hence the metrics 
-published by these new pod starts a new time series which makes tracking metrics for the application difficult. While this could be mitigated, 
+published by this new pod starts a new time series which makes tracking metrics for the application difficult. While this could be mitigated, 
 it requires some tricks on the time series collection side. It would be so much better, If we could use a StatefulSet object so my object names doesnt 
 change and hence all metrics goes to a single time series. This will be easier if StatefulSet is at feature parity with Deployments.
 2. My Container does some initial startup tasks like loading up cache or something that takes a lot of time. If we used StatefulSet, we can only go one 
-pod at a time which would result in a slow rolling update. If we did maxUnavailable for StatefulSet with a greater than 1 number, it would allow for a 
-faster rollout.
+pod at a time which would result in a slow rolling update. If StatefulSet supported maxUnavailable with value greater than 1, it would allow for a 
+faster rollout since a total of maxUnavailable number of pods could be loading up the cache at the same time.
 3. My Stateful clustered application, has followers and leaders, with followers being many more than 1. My application can tolerate many followers going 
 down at the same time. I want to be able to do faster rollouts by bringing down 2 or more followers at the same time. This is only possible if StatefulSet
 supports maxUnavailable in Rolling Updates.
-4. Sometimes i just want easier tracking of revisions of a rolling update. Deployment does it through ReplicaSets and has its own nuances. Understanding 
-that requires diving into the complicacy of hashing and how replicasets are named. Over and above that, there are some issues with hash collisions which 
-further complicate the situation(I know they were solved). StatefulSet introduced ControllerRevisions in 1.7 which I believe are easier to think and reason 
-about. They are used by DaemonSet and StatefulSet for tracking revisions. It would be so much nicer if all the use cases of Deployments can be met and we 
-could track the revisions by ControllerRevisions.
+4. Sometimes I just want easier tracking of revisions of a rolling update. Deployment does it through ReplicaSets and has its own nuances. Understanding 
+that requires diving into the complicacy of hashing and how ReplicaSets are named. Over and above that, there are some issues with hash collisions which 
```

Comment:
I feel like this point should be made clear in feedback to @kubernetes/sig-apps-api-reviews - are there any supporting statements we should have here about changes that *should* happen in deployments?